File: src/main/java/services/RepositoryService.java
package services;

import dao.RepositoryDao;
import models.Repository;

import java.util.List;

public class RepositoryService extends Service<Repository> {

    private RepositoryDao repoDao;

    public RepositoryService(int userID, String password) {
        repoDao = new RepositoryDao(userID, password);
    }

    public Repository find(int id) {
        return repoDao.findById(id);
    }

    public Repository find(String url) {
        return repoDao.findbyURL(url);
    }

    public void save(Repository repo) {
        repoDao.save(repo);
    }

    public void update(Repository repo) {
        repoDao.update(repo);
    }

    public void delete(Repository repo) {
        repoDao.delete(repo);
    }

    public Package findPackageById(int id) {
        return repoDao.findPackageById(id);
    }

    public List<Repository> getAll() {
        return repoDao.getAll();
    }
}

File: src/main/kotlin/services/RepositoryService.kt
package services

import dao.RepositoryDao
import models.Repository

class RepositoryService(userID: Int, password: String) : Service<Repository>(userID, password) {

    private var repoDao: RepositoryDao = RepositoryDao(userID, password)

    override fun reload(): RepositoryService {
        repoDao = RepositoryDao(userID, password)
        return this
    }

    override fun find(id: Int): Repository {
        return repoDao.findById(id)
    }

    fun find(url: String): Repository? {
        return repoDao.findByURL(url)
    }

    override fun save(obj: Repository) {
        repoDao.save(obj)
    }

    override fun update(obj: Repository) {
        repoDao.update(obj)
    }

    override fun delete(obj: Repository) {
        repoDao.delete(obj)
    }

    fun findPackageById(id: Int): Package {
        return repoDao.findPackageById(id)
    }

    override fun getAll(): List<Repository>? {
        return repoDao.all?.drop(1)
    }
}
